lib/widgets/drive/dynamic-list.ts

```diff
@@ -0,0 +1,248 @@
+/*
+The MIT License
+
+Copyright (c) 2015 Juan Cruz Viotti, Inc. https://github.com/jviotti.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/**
+ * @module dynamiclist
+ */
+
+import * as Bluebird from 'bluebird';
+import InquirerList = require('inquirer/lib/prompts/list');
+import UI = require('inquirer/lib/ui/baseUI');
+import * as _ from 'lodash';
+
+export = class DynamicList extends InquirerList {
+	public options: ConstructorParameters<typeof InquirerList>[0] & {
+		emptyMessage: string;
+	};
+	public ui: UI;
+
+	/**
+	 * @summary Dynamic list widget
+	 * @name DynamicList
+	 * @class
+	 * @public
+	 *
+	 * @param {Object} [options] - options
+	 * @param {Object[]} options.choices - initial choices
+	 * @param {String} options.message - widget message
+	 * @param {String} [options.emptyMessage='No options'] - message for when no choices
+	 *
+	 * @example
+	 * list = new DynamicList
+	 * 	message: 'Foo'
+	 * 	emptyMessage: 'Nothing to show'
+	 * 	choices: [
+	 * 		name: 'Foo'
+	 * 		value: 'foo'
+	 * 	]
+	 *
+	 * # Run the list widget
+	 * list.run().then (answer) ->
+	 * 	console.log(answer)
+	 *
+	 * # You can add new choices on the fly
+	 * list.addChoice
+	 * 	name: 'Bar'
+	 * 	value: 'bar'
+	 *
+	 * # We re-render to be able to see the new options
+	 * list.render()
+	 */
+	constructor(
+		options: ConstructorParameters<typeof InquirerList>[0] & {
+			emptyMessage?: string;
+		},
```

Comment:
this could ba a type as it is used twice

Reply:
It's slightly different in the two places, `emptyMessage` is required in one and not in the other